xfer.py

```diff
@@ -1,743 +1,754 @@
-    try:
-        os.makedirs(os.path.join(working_dir, "calc_work"))
-    except OSError as oe:
-        if oe.errno != errno.EEXIST:
-            raise
-
-    info = os.path.split(sys.argv[0])
-    full_exec_path = os.path.join(os.path.abspath(info[0]), info[1])
-
-    if requirements is not None:
-        with open(os.path.join(working_dir, "calc_work", "requirements.txt"), "w") as fd:
-            fd.write(requirements)
-
-    with open(os.path.join(working_dir, "xfer.dag"), "w") as fd:
-        fd.write(PARENT_DAG.format(exec_py=full_exec_path, source_prefix=source_dir,
-            dest_prefix=dest_dir, other_args="--test-mode" if test_mode else "",
-            transfer_manifest=os.path.join(dest_dir, "transfer_manifest.txt"),
-            requirements="--requirements_file=requirements.txt" if requirements is not None else "",
-            unique_id="--unique-id={}".format(unique_id) if unique_id is not None else ""))
-
-    with open(os.path.join(working_dir, "calc_work", "calc_work.sub"), "w") as fd:
-        fd.write(CALC_WORK_JOB.format(exec_py=full_exec_path, source_dir=source_dir,
-            other_args="--test-mode" if test_mode else "",
-            requirements=requirements if requirements is not None else "True",
-            unique_id=unique_id if unique_id is not None else ""))
-
-    dagman = search_path("condor_dagman")
-    if not dagman:
-        print("Unable to find the `condor_dagman` executable in the $PATH")
-        sys.exit(1)
 
-    submit_dict = {
-        b"universe":                    b"scheduler",
-        b"executable":                  dagman.encode('utf-8'),
-        b"output":                      b"xfer.dag.lib.out",
-        b"error":                       b"xfer.dag.lib.err",
-        b"log":                         b"xfer.dag.dagman.log",
-        b"remove_kill_Sig":             b"SIGUSR1",
-        b"+OtherJobRemoveRequirements": b'"DAGManJobId =?= $(cluster)"',
-        b"on_exit_remove":              b'(ExitSignal =?= 11 || (ExitCode =!= UNDEFINED && '
-                                        b'ExitCode >=0 && ExitCode <= 2))',
-        b"arguments":                   b"\"-p 0 -f -l . -Lockfile xfer.dag.lock -AutoRescue 1 -DoRescueFrom 0"
-                                        b" -Dag xfer.dag -Suppress_notification -Dagman /usr/bin/condor_dagman"
-                                        b" -CsdVersion {}\"".format(htcondor.version().replace(" ", "' '")),
-        b"environment":                 b"_CONDOR_SCHEDD_ADDRESS_FILE={};_CONDOR_MAX_DAGMAN_LOG=0;"
-                                        b"_CONDOR_SCHEDD_DAEMON_AD_FILE={};_CONDOR_DAGMAN_LOG=xfer.dag.dagman.out".format(
-            htcondor.param[b'SCHEDD_ADDRESS_FILE'], htcondor.param[b'SCHEDD_DAEMON_AD_FILE'])
+def shared_submit_descriptors(unique_id=None, requirements=None):
+    return {
+        "executable": THIS_FILE.as_posix(),
+        "My.IsTransferJob": "true",
```

Comment:
As much as I would like to use the camel case version, the START expression on the execute hosts is `START = TARGET.IS_TRANSFER_JOB`
```suggestion
        "My.Is_Transfer_Job": "true",
```